subprojects/core/src/main/java/org/gradle/execution/taskgraph/TaskExecutionGraphInternal.java

```diff
@@ -59,11 +59,14 @@ public interface TaskExecutionGraphInternal extends TaskExecutionGraph {
      */
     int size();
 
+    default boolean hasScheduledWork() {
+        return size() > 0;
+    }
+
     /**
-     * Returns all the work items in this graph scheduled for execution plus all
-     * dependencies from other builds.
+     * Returns a snapshot of currently scheduled nodes.
      */
-    void visitScheduledNodes(BiConsumer<List<Node>, Set<Node>> visitor);
+    ScheduledWork collectScheduledWork();
```

Comment:
💭  The only call-site actually needed scheduled work, and it makes sense for the task execution graph to provide one.